test/authorization_tests.js

```diff
@@ -7,276 +7,306 @@ var persona = require("../index");
+            persona_host: process.env.PERSONA_TEST_HOST || "persona",
+            persona_port: process.env.PERSONA_TEST_PORT || 80,
+            persona_scheme: process.env.PERSONA_TEST_SCHEME || "http",
+            persona_oauth_route: "/oauth/tokens/",
+            enable_debug: false
+        },
+        "redis": {
+            persona_host: process.env.PERSONA_TEST_HOST || "persona",
+            persona_port: process.env.PERSONA_TEST_PORT || 80,
+            persona_scheme: process.env.PERSONA_TEST_SCHEME || "http",
+            persona_oauth_route: "/oauth/tokens/",
+            cache: {
+                module: "redis",
+                options: {
+                    redisData: {
+                        hostname: "localhost",
+                        port: 6379
+                    }
+                }
+            },
+            enable_debug: false
+        },
+        "legacy-config-options": {
+            persona_host: process.env.PERSONA_TEST_HOST || "persona",
+            persona_port: process.env.PERSONA_TEST_PORT || 80,
+            persona_scheme: process.env.PERSONA_TEST_SCHEME || "http",
+            persona_oauth_route: "/oauth/tokens/",
+            redis_host: "localhost",
+            redis_port: 6379,
+            redis_db: 0,
             enable_debug: false
+        }
+    }, function(personaClientConfig) {
+        beforeEach(function createClientAndStubs() {
+            runBeforeEach(this.currentTest.parent.title + " " + this.currentTest.title, "authorization", true);
+
+            personaClient = persona.createClient(personaClientConfig);
+            sinon.stub(personaClient.tokenCache, "get").yields(null, null);
         });
-        sinon.stub(personaClient.redisClient, "get").yields(null, null);
-    });
 
-    afterEach(function restoreStubs() {
-        runAfterEach(this.currentTest.parent.title + " " + this.currentTest.title, "authorization", true);
-        personaClient.redisClient.get.restore();
-    });
+        afterEach(function restoreStubs() {
+            runAfterEach(this.currentTest.parent.title + " " + this.currentTest.title, "authorization", true);
+            personaClient.tokenCache.get.restore();
+        });
 
-    describe("Request authorization tests",function() {
+        describe("Request authorization tests",function() {
 
-        it("should throw an error if guid is not present", function(done) {
-            personaClient.requestAuthorization(null,"test title","some_id","some_secret",function(err,data) {
-                assert(err!=null);
-                err.should.be.a.String;
-                err.should.equal("guid, title, id and secret are required strings");
-                assert(data===null);
-                done();
+            it("should throw an error if guid is not present", function(done) {
+                personaClient.requestAuthorization(null,"test title","some_id","some_secret",function(err,data) {
+                    assert(err!=null);
```

Comment:
When I was debugging these tests - this assertion made it hard to find out the root cause. We should be replacing this with:

```javascript
if(err) {
  done(err);
}
```

This will print out the error and fail the test early.